### ISSUE TITLE
Cloudflare Action IPv6 Support

### ISSUE THREAD

**Issue by nicegamer7** [fail2ban/fail2ban#2891]
### Environment:
- Fail2Ban version (including any possible distribution suffixes): 0.11.1-2
- OS, including release name/version: Ubuntu 20.10
- [x] Fail2Ban installed via OS/distribution mechanisms
- [x] You have not applied any additional foreign patches to the codebase
- [ ] Some customizations were done to the configuration (provide details below is so)

### The issue:
I believe the Cloudflare ban action does not support IPv6. The reason I believe this is that the Cloudflare API specifies different a different "target" for `ip` and `ip6` (expand the defintions under the "configuration" section [here](https://api.cloudflare.com/#user-level-firewall-access-rule-create-access-rule)).

For `ip`, the following is the example provided:
```
{
  "target": "ip",
  "value": "198.51.100.4"
}
```

However, for `ip6`, the following:
```
{
  "target": "ip6",
  "value": "2001:DB8:100::CF"
}
```

And finally, from the `cloudflare.conf` action provided with fail2ban, only `ip` is specified:
https://github.com/fail2ban/fail2ban/blob/b8e8a87ee93985d36c8f53bf5089650b34c874e1/config/action.d/cloudflare.conf#L47

#### Steps to reproduce
I'm unable to test this as I have no access to IPv6 :(

#### Expected behavior
:(

#### Observed behavior
:(

**Comment by sebres** [comment]
No idea (I don't use cloudflare) whether cloudflare is able to correct that on its side...
If it doesn't work actually, following patch could probably help:
```diff
 actionban = curl -s -o /dev/null -X POST <_cf_api_prms> \
-             -d '{"mode":"block","configuration":{"target":"ip","value":"<ip>"},"notes":"Fail2Ban <name>"}' \
+             -d '{"mode":"block","configuration":{"target":"<target>","value":"<ip>"},"notes":"Fail2Ban <name>"}' \
             <_cf_api_url>
 ...
 actionunban = id=$(curl -s -X GET <_cf_api_prms> \
-                   "<_cf_api_url>?mode=block&configuration_target=ip&configuration_value=<ip>&page=1&per_page=1&notes=Fail2Ban%%20<name>" \
+                   "<_cf_api_url>?mode=block&configuration_target=<target>&configuration_value=<ip>&page=1&per_page=1&notes=Fail2Ban%%20<name>" \
 ...


 [Init]
+target = ip
 ...

+[Init?family=inet6]
+target = ip6
```

**Comment by nicegamer7** [comment]
Actually... come to think of it, I can test this through the command line right?

Would the following command work? (as a test)
```
fail2ban-client set {JAIL} banip {some IPv6}
```

**Comment by nicegamer7** [comment]
Alright, so I just tested it using the above command and it works. Unbanning works too. Just because I was curious, I called the ban and unban actions directly so I could see the responses. The responses indicate that Cloudflare automatically corrects `ip` to `ip6`. I could tell because the responses contain a "configuration" block, which are identical to the ones used when making the API calls, except `ip` is corrected to `ip6` (both when banning and unbanning).

However, I think the proposed change wouldn't be a bad idea, given that it is technically incorrect right now. Your call though, as its already working.

**Comment by sebres** [comment]
Sure (and take a look into fail2ban.log for possible errors). You can use reserved IP addresses, e.g.:
```
fail2ban-client set "$jail" banip 192.0.2.1   && fail2ban-client set "$jail" unbanip 192.0.2.1
fail2ban-client set "$jail" banip 2001:db8::1 && fail2ban-client set "$jail" unbanip 2001:db8::1
```

**Comment by sebres** [comment]
Oh, I was a bit late.
No errors in log also?

**Comment by nicegamer7** [comment]
Just tried those examples above and it's working as expected. No errors in the log.

**Comment by sebres** [comment]
"fixed" in 5f3f4d1e2ffcf99384c96c9800d1aa913ccce81b
